cel/canonical_eventlog.go

```diff
@@ -0,0 +1,341 @@
+// Package cel contains some basic operations of Canonical Eventlog.
+// Based on Canonical EventLog Spec (Draft) Version: TCG_IWG_CEL_v1_r0p37.
+package cel
+
+import (
+	"bytes"
+	"crypto"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/google/go-tpm/tpm2"
+)
+
+var cryptoHashToTpmAlg = map[crypto.Hash]tpm2.Algorithm{
+	crypto.SHA1:   tpm2.AlgSHA1,
+	crypto.SHA256: tpm2.AlgSHA256,
+	crypto.SHA384: tpm2.AlgSHA384,
+	crypto.SHA512: tpm2.AlgSHA512,
+}
+var tpmAlgToCryptoHash = map[tpm2.Algorithm]crypto.Hash{
+	tpm2.AlgSHA1:   crypto.SHA1,
+	tpm2.AlgSHA256: crypto.SHA256,
+	tpm2.AlgSHA384: crypto.SHA384,
+	tpm2.AlgSHA512: crypto.SHA512,
+}
```

Comment:
Is this necessary, we already have:

[tpm2.HashToAlgorithm](https://pkg.go.dev/github.com/google/go-tpm@v0.3.2/tpm2#HashToAlgorithm)

and

[tpm2.Algorithm.Hash](https://pkg.go.dev/github.com/google/go-tpm@v0.3.2/tpm2#Algorithm.Hash)

See: https://github.com/google/go-tpm/blob/90fb6247068f5197c3bd29e4085cf50124b2bece/tpm2/constants.go#L31-L89

Reply:
Ah.. I overlooked. Should use tpm2.